### PR TITLE
dumpling: support init tls without client kay/certs

### LINKED ISSUE #34530: dumpling support only use ssl ca to connect db
## Enhancement
as title

ref https://github.com/pingcap/tiflow/issues/5335

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #34530

Problem Summary:

ref https://github.com/pingcap/tiflow/issues/5335

### What is changed and how it works?
as title
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
dumpling: support init tls without client kay/certs
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
ci passed https://github.com/pingcap/tidb/runs/6365509102?check_suite_focus=true
/cc @lichunzhu  @lance6716   ptal

File: dumpling/tests/tls/run.sh [hunk 1]
```diff
@@ -27,3 +27,12 @@ run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem" --cert "$DUMPLING_TEST_DIR/dumplin
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls-schema-create.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t-schema.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t.000000000.sql"
+
+# test only oue ssl-ca without clent key and cert
+rm -rf $DUMPLING_OUTPUT_DIR
+mkdir $DUMPLING_OUTPUT_DIR
+run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem" --cert "$DUMPLING_TEST_DIR/dumpling.pem" --key "$DUMPLING_TEST_DIR/dumpling.key" --consistency none
```
Comment (lichunzhu) [inline dumpling/tests/tls/run.sh:34]: Should we delete `--cert --key` here?

**Review by lichunzhu — COMMENTED** [review]
rest LGTM

File: dumpling/tests/tls/run.sh [hunk 2]
```diff
@@ -27,3 +27,12 @@ run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem" --cert "$DUMPLING_TEST_DIR/dumplin
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls-schema-create.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t-schema.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t.000000000.sql"
+
+# test only oue ssl-ca without clent key and cert
```
Comment (lance6716) [inline dumpling/tests/tls/run.sh:31]:
```suggestion
# test only one ssl-ca without clent key and cert
```

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]
please check dumpling CI

**Comment by Ehco1996** [comment]
/hold 
le't me fix the ci

File: dumpling/tests/tls/run.sh [hunk 3]
```diff
@@ -27,3 +27,12 @@ run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem" --cert "$DUMPLING_TEST_DIR/dumplin
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls-schema-create.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t-schema.sql"
 file_should_exist "$DUMPLING_OUTPUT_DIR/tls.t.000000000.sql"
+
+# test only use ssl-ca without clent key and cert
+rm -rf $DUMPLING_OUTPUT_DIR
+mkdir $DUMPLING_OUTPUT_DIR
+run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem"  --consistency none
```
Comment (lichunzhu) [inline dumpling/tests/tls/run.sh:34]:
```suggestion
run_dumpling --ca "$DUMPLING_TEST_DIR/ca.pem" --consistency none
```

**Comment by Ehco1996** [comment]
/unhold

ci finally passed https://github.com/pingcap/tidb/runs/6381338636?check_suite_focus=true

😵😵😵 it's about some permiss issue see https://github.com/pingcap/tidb/pull/34483/commits/c0759cb22e4239e00e6a3f49cf9eb9153deb7ea9

**Pull request merged**